url9ecf1e8297_nhamidi_ninja_tgz-pJ8-net_fec_openrq_test_TestRunnerJ8/src/EmetteurThreadMulti.java

```diff
@@ -0,0 +1,42 @@
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+
+public class EmetteurThreadMulti extends Thread {
+    
+    InetAddress groupeIP;
+    private int port;
+    private byte[] ACK_NACK;
+    MulticastSocket socketEmission;
+    
+    EmetteurThreadMulti(InetAddress groupeIP, int port) throws Exception {
+	this.groupeIP = groupeIP;
+	this.port = port;
+	this.ACK_NACK = new byte[Utils.IntegerSize];
+	socketEmission = new MulticastSocket();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *socketEmission*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).